packages/boot/src/__tests__/acceptance/crud-rest-builder.acceptance.ts

```diff
@@ -0,0 +1,168 @@
+      'controllers.ProductController',
+    );
+  });
+
+  it('uses bound repository class if it exists', async () => {
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/product.model.js'),
+      'models/product.model.js',
+    );
+
+    await sandbox.writeTextFile(
+      'model-endpoints/product.rest-config.js',
+      `
+const {Product} = require('../models/product.model');
+module.exports = {
+  model: Product,
+  pattern: 'CrudRest',
+  dataSource: 'db',
+  basePath: '/products',
+};
+      `,
+    );
+
+    const ProductRepository = defineCrudRepositoryClass(Product);
+    app.repository(ProductRepository);
+
+    const binding = app.getBinding('repositories.ProductRepository');
+    expect(binding.key).to.eql('repositories.ProductRepository');
+
+    // Boot & start the application
+    await app.boot();
+    await app.start();
+
+    expect(app.getBinding('repositories.ProductRepository')).to.eql(binding);
+
+    expect(app.getBinding('controllers.ProductController').key).to.eql(
+      'controllers.ProductController',
+    );
+  });
+
+  it('throws if there is no base path in the config', async () => {
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/product.model.js'),
+      'models/product.model.js',
+    );
+
+    await sandbox.writeTextFile(
+      'model-endpoints/product.rest-config.js',
+      `
+const {Product} = require('../models/product.model');
+module.exports = {
+  model: Product,
+  pattern: 'CrudRest',
+  dataSource: 'db',
+};
+      `,
+    );
+
+    // Boot the application
+    await expect(app.boot()).to.be.rejectedWith(
+      /Missing required field "basePath" in configuration for model Product./,
+    );
+  });
+
+  it('throws if a Model is used instead of an Entity', async () => {
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/no-entity.model.js'),
+      'models/no-entity.model.js',
+    );
+
+    await sandbox.writeTextFile(
+      'model-endpoints/no-entity.rest-config.js',
+      `
+const {NoEntity} = require('../models/no-entity.model');
+module.exports = {
+  model: NoEntity,
+  pattern: 'CrudRest',
+  dataSource: 'db',
+  basePath: '/no-entities',
+};
+      `,
+    );
+
+    // Boot the application
+    await expect(app.boot()).to.be.rejectedWith(
+      /CrudRestController requires an Entity, Models are not supported/,
+    );
+  });
+
+  class BooterApp extends BootMixin(RepositoryMixin(RestApplication)) {
+    constructor(options?: ApplicationConfig) {
+      super(options);
+      this.projectRoot = sandbox.path;
+      this.booters(ModelApiBooter);
+      this.component(CrudRestComponent);
+    }
+  }
+
+  async function givenAppWithDataSource() {
+    app = new BooterApp({
+      rest: givenHttpServerConfig(),
+    });
+    app.dataSource(new juggler.DataSource({connector: 'memory'}), 'db');
+  }
+
+  async function stopApp() {
+    try {
+      await app.stop();
+    } catch (err) {
+      // application is booting
+    }
+  }
+});
```

Comment:
The tests look legit to me 👍 I don't know if we support configurations yet. It'd be good if we can have a test for it.